sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -209,7 +186,7 @@ export class AppConfigurationClient {
         entity: keyValue,
         ...newOptions,
       });
-      return transformKeyValueResponse(originalResponse);
+      return transformKeyValueResponse(originalResponse) as AddConfigurationSettingResponse;
```

Comment:
The cast was needed here because the generated code doesn't think it has `_response` in it.
![image](https://user-images.githubusercontent.com/10452642/161171368-31e41695-e17e-424a-925a-4ac67e6927ad.png)
![image](https://user-images.githubusercontent.com/10452642/161171515-08ab81f0-616a-4f1a-9d9d-8fdb5a023060.png)

Even if I update `transformKeyValueResponse` method somehow, I'll have to use `"as {_response: <bla>}"` at some level.

Reply:
What do you suggest?

Reply:
I think it would be nice to use an assertion here:

```ts
interface WithResponse {
    _response: {
        // the real shape here
    }
}

function assertResponse<T extends object>(result: T): asserts result is T & WithResponse {
    if (!Object.prototype.hasOwnProperty.call(result, "_response")) {
        // throw? log?
    }
}

function exampleMethod(): { foo: boolean, _response: unknown} {
    const foo = {foo: true};
    assertResponse(foo);
    return foo;

}
```

Reply:
Updated